autoscaler/controllers/actions/piimasking_controller.go

```diff
@@ -0,0 +1,183 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package actions
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	actionv1 "github.com/odigos-io/odigos/api/actions/v1alpha1"
+	v1 "github.com/odigos-io/odigos/api/odigos/v1alpha1"
+	"github.com/odigos-io/odigos/common"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+var piiMaskingSupportedSignals = map[common.ObservabilitySignal]struct{}{
+	common.TracesObservabilitySignal: {},
+}
+
+type PiiMaskingReconciler struct {
+	client.Client
+	Scheme *runtime.Scheme
+}
+
+func (r *PiiMaskingReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	logger := log.FromContext(ctx)
+	logger.V(0).Info("Reconciling PiiMasking action")
+
+	action := &actionv1.PiiMasking{}
+	err := r.Get(ctx, req.NamespacedName, action)
+	if err != nil {
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+
+	for _, signal := range action.Spec.Signals {
+
+		if _, ok := piiMaskingSupportedSignals[signal]; !ok {
+
+			err = fmt.Errorf("unsupported signal: %s", signal)
+			logger.V(0).Error(nil, "PiiMasking action invalid configration")
```

Comment:
```suggestion
			logger.V(0).Error(err, "PiiMasking action invalid configration")
```

Reply:
Nice catchup, I see in the probablistic it's alsi like this.. changing there too